core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -266,11 +264,76 @@ public static FileIO loadFileIO(
           String.format("Cannot initialize FileIO, %s does not implement FileIO.", impl), e);
     }
 
-    if (fileIO instanceof Configurable) {
-      ((Configurable) fileIO).setConf(hadoopConf);
-    }
+    configureHadoopConf(fileIO, hadoopConf);
 
     fileIO.initialize(properties);
     return fileIO;
   }
+
+  /**
+   * Dynamically detects whether an object is a Hadoop Configurable and calls setConf.
+   * @param maybeConfigurable an object that may be Configurable
+   * @param conf a Configuration
+   */
+  @SuppressWarnings("unchecked")
+  public static void configureHadoopConf(Object maybeConfigurable, Object conf) {
```

Comment:
does this need to be public?

Reply:
It seems useful? I think we could make it private though.

Reply:
it is unclear to me how users would use it outside `loadCatalog` and `loadFileIO`, but maybe I am missing some use cases, so it's up to you.